components/nexus-httpclient/src/main/java/org/sonatype/nexus/httpclient/config/NonProxyHostsValidator.java

```diff
@@ -26,6 +27,8 @@
 public class NonProxyHostsValidator
     extends ConstraintValidatorSupport<NonProxyHosts, String[]>
 {
+  private static final Pattern NON_PROXY_HOST_PATTERN = Pattern.compile("^((\\p{IsAlphabetic}|\\d|\\-|\\_|\\*)+\\.?)*$", Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
You could always check the name after removing all the '*' characters? BTW, Guava has various utility methods to validate IP addresses and domain names, such as

http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/net/InetAddresses.html#isInetAddress(java.lang.String)

for numeric IPs and

http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/net/InternetDomainName.html#isValid(java.lang.String)

for domain names, neither of which do any lookup.

Note: I also saw http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/net/HostSpecifier.html#isValid(java.lang.String) but that also enforces that a domain name must end with a public suffix which I don't think we want.